sos/plugins/kubernetes.py

```diff
@@ -15,40 +15,130 @@
-                for line in result['output'].splitlines()[1:]:
-                    pod_name = line.split(" ")[0]
-                    self.add_cmd_output([
-                        "{0} log {1}".format("kubectl", pod_name)
-                    ])
+        svcs = [
+            'kubelet', 'kube-apiserver', 'kube-proxy', 'kube-scheduler',
+            'kube-controller-manager'
+        ]
+
+        for svc in svcs:
+            self.add_cmd_output("journalctl -u {}".format(svc))
+
+        # We can only grab kubectl output from the master
+        if self.check_is_master():
+            kube_bin = "kubectl "
+            if path.exists('/etc/origin/master/admin.kubeconfig'):
+                kube_bin += "--config=/etc/origin/master/admin.kubeconfig"
+
+            kube_get = "get -o json "
+            if self.get_option('all'):
+                kube_get += "--all-namespaces=true"
+
+            for cmd in ['version', 'config view']:
+                self.add_cmd_output("{} {}".format(kube_bin, cmd))
+
+            resources = [
+                'events', 'limitrange', 'pods', 'pvc', 'rc',
+                'resourcequota', 'services'
+            ]
+
+            for res in resources:
+                self.add_cmd_output(
+                    "{} {} {}".format(
+                        kube_bin,
+                        kube_get,
+                        res
+                    )
+                )
+            resources.remove('events')
+            # nodes and pvs are not namespaced, must pull separately
+            self.add_cmd_output([
+                "{} get -o json nodes".format(kube_bin),
+                "{} get -o json pv".format(kube_bin)
+                ]
+            )
+
+            if self.get_option('describe'):
+                resources = resources + ['nodes', 'namespaces']
+                for res in resources:
+                    r = self.get_command_output(
+                        "{} get {}".format(
+                            kube_bin,
+                            res
+                        )
+                    )
+                    if r['status'] == 0:
+                        k_list = [k.split()[0] for k in
+                                  r['output'].splitlines()[1:]
+                                  ]
+                        for k in k_list:
+                            self.add_cmd_output(
+                                "{} describe {} {}".format(
+                                    kube_bin,
+                                    res,
+                                    k
+                                )
+                            )
+
+            if self.get_option('podlogs'):
+                r = self.get_command_output(
+                    "{} get pods".format(kube_bin)
+                )
+                if r['status'] == 0:
+                    pods = [p.split()[0] for p in
+                            r['output'].splitlines()[1:]
+                            ]
```

Comment:
Ditto - I think this would be must cleaner in its own function. That way you can just `return` when the command produces no output and you're not stuck in ever increasing levels of indentation.